packages/adblocker/src/lists.ts

```diff
@@ -136,13 +137,20 @@ export function f(strings: TemplateStringsArray): NetworkFilter | CosmeticFilter
 export function parseFilters(
   list: string,
   config: Partial<Config> = new Config(),
-): { networkFilters: NetworkFilter[]; cosmeticFilters: CosmeticFilter[] } {
+): {
+  networkFilters: NetworkFilter[];
+  cosmeticFilters: CosmeticFilter[];
+  preprocessors: Preprocessor[];
+} {
   config = new Config(config);
 
   const networkFilters: NetworkFilter[] = [];
   const cosmeticFilters: CosmeticFilter[] = [];
   const lines = list.split('\n');
 
+  const conditions: Map<string, Preprocessor> = new Map();
+  const preprocessorStack: { instance: Preprocessor; negated: boolean }[] = [];
```

Comment:
Why not having a simple list of `Preprocessor` objects? (I see that we merge different blocks with the same condition, I guess that's the reason)

Reply:
Only that is the reason. I don't like to iterate every preprocessor to find out if there's a preprocessor with same condition. However, this behavior can be changed anytime.

Reply:
This was updated to accept list of filter ids in parsing time of preprocessor. Thus there's no need to keep this at parsing code.

The implementation was changed on this. I'm closing this as this is now outdated.